bot/exts/valentines/markov_poem_generator.py

```diff
@@ -0,0 +1,513 @@
+
+            You may also use our existing rhyme scheme templates:
+            {chr(10).join((k + " --- " + v) for k, v in templates.items())}""",
+        brief="Gives the user a love poem made with a markov chain."
+    )
+    async def markov_poem(
+        self,
+        ctx: commands.Context,
+        rhyme_scheme: str
+    ) -> None:
+        """
+        Gives the user a love poem based on a rhyme_scheme.
+
+        The user's `rhyme_scheme` is processed before being placed into the
+        poem generation process.
+
+        In this code, a unit is defined by a single character of the scheme. If
+        two or more units are the same, they are meant to rhyme (i.e their last
+        words rhyme).
+
+        The time taken to process the command is recorded and shown in the
+        footer of the embed.
+        """
+        scheme = rhyme_scheme.lower()
+        scheme = self.templates.get(scheme, scheme)
+        unit_count = self._get_unit_count(scheme)
+        time_start = datetime.now()
+
+        await self.send_markov_poem(ctx, scheme, unit_count, time_start)
+
+    async def cog_command_error(
+        self,
+        ctx: commands.Context,
+        error: Exception
+    ) -> None:
+        """Handles Discord errors and exceptions."""
+        if isinstance(error, RhymingSentenceNotFound):
+            error.handled = True
+            await self.send_markov_poem(
+                ctx, error.scheme, error.unit_count, error.time_start,
+                is_first_error=False
+            )
+        elif isinstance(error, RhymeAPIUnresponsive):
+            error.handled = True
+            embed = Embed(
+                title="Uh oh... Markov Poem can't give you a poem...",
+                color=Colours.soft_red,
+                description=f"""
+                    I am sorry {ctx.author.mention}, but it looks like the API
+                    that is used to help make your poems is down...
+                    Please come back another time.
+                    """
+            )
+            await ctx.send(embed=embed)
+        elif isinstance(error, MakeShortSentenceRanOut):
+            error.handled = True
+            embed = Embed(
+                title="Uh oh... Markov Poem can't give you a poem...",
+                color=Colours.soft_red,
+                description=f"""
+                    Apologies {ctx.author.mention},
+                    but it appears that we have encountered a rare bug.
+                    It might be the case that your poem is too long,
+                    please try again.
+                    """
+            )
+            await ctx.send(embed=embed)
+        elif isinstance(error, commands.errors.MaxConcurrencyReached):
+            error.handled = True
+            await ctx.send(f"Please be patient {ctx.author.mention},"
+                           " we are currently working on your poem!")
+
+
+def setup(bot: commands.Bot) -> None:
+    """Poem generator cog load."""
```

Comment:
```suggestion
    """Load the Poem Generator cog."""
```
The docstrings for the setup function have been made uniform across the repo in #718 , let's follow that here.